js/__internal/grids/grid_core/keyboard_navigation/module.ts

```diff
@@ -2041,52 +2052,59 @@ export class KeyboardNavigationController extends modules.ViewController {
     return args;
   }
 
-  _fireFocusedCellChanged($cellElement, prevCellIndex, prevRowIndex) {
-    const that = this;
-    const dataController = that._dataController;
-    const columnIndex = that._rowsView.getCellIndex($cellElement);
-    const rowIndex = this._getRowIndex($cellElement && $cellElement.parent());
+  _fireFocusedCellChanged($cell: dxElementWrapper): void {
+    const columnIndex = this._rowsView.getCellIndex($cell);
+    const rowIndex = this._getRowIndex($cell?.parent());
```

Comment:
minor: if we need to use `?` with `$cell`, let's mark it with `| undefined`

Also let's add https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-unnecessary-condition.md to linter

Reply:
Discussed.

Agree with you.
Fixed.